File: go_app/entities/sapien.go
package entities

import "time"

type Sapien struct {
	CommonAccount
	XToken        string
	DCToken       string
	Points        float64   `gorm:"default:0"`
	UnderPoints   float64   `gorm:"default:0"`
	DailyAt       time.Time `gorm:"index"`
	IsBindSocial  bool      `gorm:"default:false"`
	Email         string
	EmailPassword string
}

func (*Sapien) TableName() string {
	return "sapien"
}

File: ts_app/schema/sapien.ts
import {
  boolean,
  datetime,
  decimal,
  mysqlTable,
  varchar,
} from "drizzle-orm/mysql-core";
import { Account, baseColumnsV2 } from "./base_models";

export const SapienTable = mysqlTable(`sapien`, {
  ...baseColumnsV2(),
  points: decimal("points"),
  under_points: decimal("under_points"),
  daily_at: datetime("daily_at"),
  x_token: varchar("x_token", { length: 255 }),
  dc_token: varchar("dc_token", { length: 255 }),
  is_bind_social: boolean("is_bind_social"),
  email: varchar("email", { length: 256 }),
  email_password: varchar("email_password", { length: 256 }),
});

export type SapienAccount = Account<typeof SapienTable>;
